eng/tools/lint-diff/src/lint-diff.ts

```diff
@@ -0,0 +1,303 @@
+    );
+
+    changedFileAndTagsMap.set(readmeAndTags.readme, dedupedTags);
+  }
+
+  // TODO:
+  for (const file of changedFileAndTagsMap.keys()) {
+    const changedFilePath = `${targetPath}/${file}`;
+    console.log(`Linting ${changedFilePath}`);
+
+    let openApiType = await getOpenapiType(changedFilePath);
+
+    // From momentOfTruth.ts:executeAutoRestWithLintDiff
+    // This is a quick workaround for https://github.com/Azure/azure-sdk-tools/issues/6549
+    // We override the openapi-subtype with the value of openapi-type,
+    // to prevent LintDiff from reading openapi-subtype from the AutoRest config file (README)
+    // and overriding openapi-type with it.
+    let openApiSubType = openApiType;
+
+    // TODO: Is there a more syntactically graceful way to handle empty tags?
+    for (const tag of changedFileAndTagsMap.get(file) || [null]) {
+      // TODO: See tag specified in momentOfTruth.ts:executeAutoRestWithLintDiff
+      let tagArg = tag ? `--tag=${tag} ` : "";
+
+      let autorestCommand =
+        `npm exec -- autorest ` +
+        `--v3 ` +
+        `--spectral ` +
+        `--azure-validator ` +
+        `--semantic-validator=false ` +
+        `--model-validator=false ` +
+        `--message-format=json ` +
+        `--openapi-type=${openApiType} ` +
+        `--openapi-subtype=${openApiSubType} ` +
+        `--use=${validator} ` +
+        `${tagArg} ` +
+        `${changedFilePath}`;
+
+      console.log(`autorest command: ${autorestCommand}`);
+      let lintDiffResult = await executeCommand(autorestCommand);
+
+      if (lintDiffResult.error) {
+        console.error("Error running autorest", lintDiffResult.error);
+        process.exit(1);
+      }
+
+      // console.log("Lint diff result:", lintDiffResult.stdout);
+      console.log("Results would be written to: ", outFile);
+    }
+  }
+}
+
+async function executeCommand(
+  command: string,
+  cwd: string = ".",
+): Promise<{ error: Error | null; stdout: string; stderr: string }> {
+  return new Promise((resolve) => {
+    exec(
+      command,
+      { cwd, encoding: "utf-8", maxBuffer: MAX_EXEC_BUFFER },
+      (error, stdout, stderr) => {
+        resolve({ error, stdout, stderr });
+      },
+    );
+  });
+}
+
+async function readFileList(changedFilesPath: string): Promise<string[]> {
+  const data = await readFile(changedFilesPath, { encoding: "utf-8" });
+  return data
+    .split("\n")
+    .map((line) => line.trim())
+    .filter((line) => line.length > 0);
+}
+
+/**
+ * Get the list of readme files that are affected by the changed files by
+ * searching for readme files in each higher directory up to the "specification/"
+ * directory.
+ *
+ * Mostly stands in for getReadmeDiffs in momentOfTruth.ts
+ *
+ * TODO: TESTS
+ * @param changedFiles List of changed files.
+ */
+async function getAffectedReadmeFiles(changedFiles: string[], repoRoot: string): Promise<string[]> {
```

Comment:
Might be worth making the rest of the file shared code, if we think other tools may want similar file lists.

Reply:
Covered in https://github.com/Azure/azure-rest-api-specs/issues/33302